tests/components/modbus/test_sensor_sizes.py

```diff
@@ -0,0 +1,864 @@
+"""The tests for the Modbus sensor component."""
+
+import pytest
+
+from homeassistant.components.modbus.const import (
+    CONF_DATA_TYPE,
+    CONF_MAX_VALUE,
+    CONF_MIN_VALUE,
+    CONF_NAN_VALUE,
+    CONF_PRECISION,
+    CONF_REGISTER_SIZE_BYTES,
+    CONF_SCALE,
+    CONF_SLAVE_COUNT,
+    CONF_SWAP,
+    CONF_SWAP_BYTE,
+    CONF_SWAP_WORD,
+    CONF_SWAP_WORD_BYTE,
+    CONF_VIRTUAL_COUNT,
+    CONF_ZERO_SUPPRESS,
+    DataType,
+)
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.const import (
+    CONF_ADDRESS,
+    CONF_COUNT,
+    CONF_NAME,
+    CONF_OFFSET,
+    CONF_SENSORS,
+    CONF_STRUCTURE,
+    CONF_UNIQUE_ID,
+    STATE_UNAVAILABLE,
+    STATE_UNKNOWN,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from .conftest import TEST_ENTITY_NAME
+
+ENTITY_ID = f"{SENSOR_DOMAIN}.{TEST_ENTITY_NAME}".replace(" ", "_")
+SLAVE_UNIQUE_ID = "ground_floor_sensor"
+
+
+@pytest.mark.parametrize(
+    "do_config",
+    [
+        {
+            CONF_SENSORS: [
+                {
+                    CONF_NAME: TEST_ENTITY_NAME,
+                    CONF_ADDRESS: 201,
+                },
+            ],
+        }
+    ],
+)
+@pytest.mark.parametrize(
+    ("config_addon", "expected_count"),
+    [
+        (
+            {
+                CONF_DATA_TYPE: DataType.UINT16,
+                CONF_REGISTER_SIZE_BYTES: 2,
+            },
+            1,
+        ),
+        (
+            {
+                CONF_DATA_TYPE: DataType.UINT32,
+                CONF_REGISTER_SIZE_BYTES: 2,
+            },
+            2,
+        ),
+        (
+            {
+                CONF_DATA_TYPE: DataType.UINT64,
+                CONF_REGISTER_SIZE_BYTES: 2,
+            },
+            4,
+        ),
+        (
+            {
+                CONF_DATA_TYPE: DataType.UINT32,
+                CONF_REGISTER_SIZE_BYTES: 4,
+            },
+            1,
+        ),
+        (
+            {
+                CONF_DATA_TYPE: DataType.FLOAT64,
+                CONF_REGISTER_SIZE_BYTES: 4,
+            },
+            2,
+        ),
+        (
+            {
+                CONF_DATA_TYPE: DataType.FLOAT64,
+                CONF_REGISTER_SIZE_BYTES: 8,
+            },
+            1,
+        ),
+        (
+            {
+                CONF_DATA_TYPE: DataType.CUSTOM,
```

Comment:
CUSTOM with register_size: 8 is missing.

Reply:
added